applications/RomApplication/custom_strategies/rom_builder_and_solver.h

```diff
@@ -676,6 +410,347 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
+        RomSystemMatrixType &A,
+        RomSystemVectorType &b)
+    {
+        KRATOS_TRY
+
+        // Define a dense matrix to hold the reduced problem
+        A = ZeroMatrix(mNumberOfRomModes, mNumberOfRomModes);
+        b = ZeroVector(mNumberOfRomModes);
+
+        // Build the system matrix by looping over elements and conditions and assembling to A
+        KRATOS_ERROR_IF(!pScheme) << "No scheme provided!" << std::endl;
+
+        // Get ProcessInfo from main model part
+        const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+        // Assemble all entities
+        const auto assembling_timer = BuiltinTimer();
+
+        using SystemSumReducer = CombinedReduction<NonTrivialSumReduction<RomSystemMatrixType>, NonTrivialSumReduction<RomSystemVectorType>>;
+        AssemblyPrealocation prealloc(mNumberOfRomModes);
+
+        auto& elements = mHromSimulation ? mSelectedElements : rModelPart.Elements();
+        if(!elements.empty())
+        {
+            std::tie(A, b) =
+            block_for_each<SystemSumReducer>(elements, prealloc, 
+                [&](Element& r_element, AssemblyPrealocation& thread_prealloc)
+            {
+                return CalculateLocalContribution(r_element, thread_prealloc, *pScheme, r_current_process_info);
+            });
+        }
+
+        auto& conditions = mHromSimulation ? mSelectedConditions : rModelPart.Conditions();
+        if(!conditions.empty())
+        {
+            RomSystemMatrixType Aconditions;
+            RomSystemVectorType bconditions;
+
+            std::tie(Aconditions, bconditions) =
+            block_for_each<SystemSumReducer>(conditions, prealloc, 
+                [&](Condition& r_condition, AssemblyPrealocation& thread_prealloc)
+            {
+                return CalculateLocalContribution(r_condition, thread_prealloc, *pScheme, r_current_process_info);
+            });
+
+            A += Aconditions;
+            b += bconditions;
+        }
+
+        // Syncronizing
+        constexpr IndexType root_rank = 0;
+        for(IndexType i=0; i<mNumberOfRomModes; ++i) {
+            b(i) = rModelPart.GetCommunicator().GetDataCommunicator().Sum(b(i), root_rank);
+            for(IndexType j=0; j<mNumberOfRomModes; ++j) {
+                A(i,j) = rModelPart.GetCommunicator().GetDataCommunicator().Sum(A(i,j), root_rank);
+            }
+        }
+
+        KRATOS_INFO_IF("ROMBuilderAndSolver", (this->GetEchoLevel() > 0 && rModelPart.GetCommunicator().MyPID() == 0)) << "Build time: " << assembling_timer.ElapsedSeconds() << std::endl;
+        KRATOS_INFO_IF("ROMBuilderAndSolver", (this->GetEchoLevel() > 2 && rModelPart.GetCommunicator().MyPID() == 0)) << "Finished parallel building" << std::endl;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * Solves reduced system of equations and broadcasts it
+     */
+    void SolveROM(
+        ModelPart &rModelPart,
+        RomSystemMatrixType &Arom,
+        RomSystemVectorType &brom,

```

Comment:
```suggestion
        RomSystemMatrixType &rA,
        RomSystemVectorType &rb,
```
I'd use the same naming convention as in `BuildROM`.